### PR TITLE
Refine conditions for displaying the install azd prompt

### LINKED ISSUE #1713: Update "install azd" notification in VS Code to be more descriptive
Today, the prompt just says that we didn't detect azd and asked the user if they want to install. We should make this more descriptive/add a value prop so people know why they'd want it.

### PR DESCRIPTION
**Pull request by bwateratmsft**
Fixes #1713 as well.

### CODE REVIEW

File: ext/vscode/src/utils/azureDevCli.ts [hunk 1]
```diff
@@ -171,26 +164,20 @@ function normalize(env: NodeJS.ProcessEnv): Environment {
 }
 
 function azdNotInstalledMsg(): string {
-    return localize("azure-dev.utils.azd.notInstalled", "Azure Developer CLI is not installed. Would you like to install it?.");
+    return localize("azure-dev.utils.azd.notInstalled", "Azure Developer CLI is not installed. [Learn More](https://aka.ms/azd-install). Would you like to install it?");
```
Comment (bwateratmsft) [inline ext/vscode/src/utils/azureDevCli.ts:167]: TODO: Need final text from @savannahostrowski / @gkulin

Comment (savannahostrowski) [reply ext/vscode/src/utils/azureDevCli.ts:167]: Can we put the learn more link last in the phrase? Otherwise, LGTM.

Comment (bwateratmsft) [reply ext/vscode/src/utils/azureDevCli.ts:167]: How's this? 
image

Comment (savannahostrowski) [reply ext/vscode/src/utils/azureDevCli.ts]: Yep - ship it!

**Review by karolz-ms — APPROVED** [review]

**Pull request merged**